<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The Irritable Bowel Syndrome-associated Microbiome â€¢ Analyses based on bugsigdb.org</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The Irritable Bowel Syndrome-associated Microbiome">
<meta property="og:description" content="bugSigSimple">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Analyses based on bugsigdb.org</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/c-section_meconium_shaimaa.html">Delivery Mode and the Meconium Microbiome</a>
    </li>
    <li>
      <a href="../articles/capstoneanalysis_clare.html">COVID-19 associated microbiome</a>
    </li>
    <li>
      <a href="../articles/capstoneanalysis_fatima.html">An analysis of Most frequent taxa in Major Depression and Bipolar Disorder</a>
    </li>
    <li>
      <a href="../articles/capstoneanalysis_kweku.html">The Irritable Bowel Syndrome-associated Microbiome</a>
    </li>
    <li>
      <a href="../articles/fieldworkanalysis_samara.html">The Endometriosis-associated Microbiome</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/bugSigSimple" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="capstoneanalysis_kweku_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="capstoneanalysis_kweku_files/kePrint-0.0.1/kePrint.js"></script><link href="capstoneanalysis_kweku_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The Irritable Bowel Syndrome-associated Microbiome</h1>
                        <h4 data-toc-skip class="author">Kweku Amoo</h4>
            
            <h4 data-toc-skip class="date">8/17/2021</h4>
      
      
      <div class="hidden name"><code>capstoneanalysis_kweku.rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="analysis-objectives">Analysis objectives<a class="anchor" aria-label="anchor" href="#analysis-objectives"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Import, recode, and subset data from bugsigdb.org<br>
</li>
<li>Create a table of studies</li>
<li>Create a clustered heatmap showing similarity of signatures from independent studies</li>
<li>Calculate the frequency of appearance of each taxa in independent signatures, and identify the most frequently reported taxa</li>
<li>Estimate the probability of the most frequently identified taxa occuring so frequently by chance</li>
</ol>
</div>
<div class="section level2">
<h2 id="packages-installation">Packages installation<a class="anchor" aria-label="anchor" href="#packages-installation"></a>
</h2>
<p>Install packages (not evaluated in vignette)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"devtools"</span>, <span class="st">"tidyverse"</span>, <span class="st">"kableExtra"</span>, <span class="st">"gt"</span>, <span class="st">"glue"</span><span class="op">)</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"waldronlab/bugSigSimple"</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"waldronlab/BugSigDBStats"</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"waldronlab/bugsigdbr"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-import-recoding-and-subset">Data import, recoding, and subset<a class="anchor" aria-label="anchor" href="#data-import-recoding-and-subset"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://waldronlab.io/bugSigSimple" class="external-link">bugSigSimple</a></span><span class="op">)</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">bugsigdbr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bugsigdbr/man/importBugSigDB.html" class="external-link">importBugSigDB</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 2098   48</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "Study"                      "Study design"              </span>
<span class="co">##  [3] "PMID"                       "DOI"                       </span>
<span class="co">##  [5] "URL"                        "Authors"                   </span>
<span class="co">##  [7] "Title"                      "Journal"                   </span>
<span class="co">##  [9] "Year"                       "Experiment"                </span>
<span class="co">## [11] "Location of subjects"       "Host species"              </span>
<span class="co">## [13] "Body site"                  "UBERON ID"                 </span>
<span class="co">## [15] "Condition"                  "EFO ID"                    </span>
<span class="co">## [17] "Group 0 name"               "Group 1 name"              </span>
<span class="co">## [19] "Group 1 definition"         "Group 0 sample size"       </span>
<span class="co">## [21] "Group 1 sample size"        "Antibiotics exclusion"     </span>
<span class="co">## [23] "Sequencing type"            "16S variable region"       </span>
<span class="co">## [25] "Sequencing platform"        "Statistical test"          </span>
<span class="co">## [27] "Significance threshold"     "MHT correction"            </span>
<span class="co">## [29] "LDA Score above"            "Matched on"                </span>
<span class="co">## [31] "Confounders controlled for" "Pielou"                    </span>
<span class="co">## [33] "Shannon"                    "Chao1"                     </span>
<span class="co">## [35] "Simpson"                    "Inverse Simpson"           </span>
<span class="co">## [37] "Richness"                   "Signature page name"       </span>
<span class="co">## [39] "Source"                     "Curated date"              </span>
<span class="co">## [41] "Curator"                    "Revision editor"           </span>
<span class="co">## [43] "Description"                "Abundance in Group 1"      </span>
<span class="co">## [45] "MetaPhlAn taxon names"      "NCBI Taxonomy IDs"         </span>
<span class="co">## [47] "State"                      "Reviewer"</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Registered S3 methods overwritten by 'readr':</span>
<span class="co">##   method                    from </span>
<span class="co">##   as.data.frame.spec_tbl_df vroom</span>
<span class="co">##   as_tibble.spec_tbl_df     vroom</span>
<span class="co">##   format.col_spec           vroom</span>
<span class="co">##   print.col_spec            vroom</span>
<span class="co">##   print.collector           vroom</span>
<span class="co">##   print.date_names          vroom</span>
<span class="co">##   print.locale              vroom</span>
<span class="co">##   str.col_spec              vroom</span></code></pre>
<pre><code><span class="co">## â”€â”€ <span style="font-weight: bold;">Attaching packages</span> â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.3.1 â”€â”€</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">ggplot2</span> 3.3.5     <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">purrr  </span> 0.3.4</span>
<span class="co">## <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">tibble </span> 3.1.6     <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">dplyr  </span> 1.0.8</span>
<span class="co">## <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">tidyr  </span> 1.2.0     <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">stringr</span> 1.4.0</span>
<span class="co">## <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">readr  </span> 2.1.2     <span style="color: #00BB00;">âœ”</span> <span style="color: #0000BB;">forcats</span> 0.5.1</span></code></pre>
<pre><code><span class="co">## â”€â”€ <span style="font-weight: bold;">Conflicts</span> â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€</span>
<span class="co">## <span style="color: #BB0000;">âœ–</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span> masks <span style="color: #0000BB;">stats</span>::filter()</span>
<span class="co">## <span style="color: #BB0000;">âœ–</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>    masks <span style="color: #0000BB;">stats</span>::lag()</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">condition_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"irritable bowel syndrome"</span><span class="op">)</span>
<span class="va">dat_condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetByCondition.html">subsetByCondition</a></span><span class="op">(</span><span class="va">dat</span>, <span class="va">condition_of_interest</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>comparison1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">`Group 0 name`</span>, <span class="va">`Group 1 name`</span>, sep <span class="op">=</span> <span class="st">" vs "</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="table-of-studies">Table of studies<a class="anchor" aria-label="anchor" href="#table-of-studies"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">bugSigSimple</span><span class="fu">::</span><span class="fu"><a href="../reference/createStudyTable.html">createStudyTable</a></span><span class="op">(</span><span class="va">dat_condition</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 9 Ã— 5</span></span>
<span class="co">##   Study           Condition                Cases Controls `Study Design`        </span>
<span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span>
<span class="co">## <span style="color: #BCBCBC;">1</span> Biagi 2011      irritable bowel syndrome    62       46 case-control,prospectâ€¦</span>
<span class="co">## <span style="color: #BCBCBC;">2</span> Carroll 2012    irritable bowel syndrome    23       23 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">3</span> Duboc 2012      irritable bowel syndrome    14       18 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">4</span> Kerckhoffs 2009 irritable bowel syndrome    41       26 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">5</span> Maccaferri 2012 irritable bowel syndrome    19       24 randomized controlledâ€¦</span>
<span class="co">## <span style="color: #BCBCBC;">6</span> Mei 2021        irritable bowel syndrome    30       30 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">7</span> Saulnier 2011   irritable bowel syndrome    22       22 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">8</span> Shukla 2015     irritable bowel syndrome    47       30 case-control          </span>
<span class="co">## <span style="color: #BCBCBC;">9</span> Tana 2010       irritable bowel syndrome    26       26 case-control</span></code></pre>
</div>
<div class="section level3">
<h3 id="taxon-frequency-tables-by-body-site">Taxon frequency tables by body site<a class="anchor" aria-label="anchor" href="#taxon-frequency-tables-by-body-site"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gut_sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dat_condition</span>,
                   <span class="va">`Body site`</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feces,mucosa of small intestine"</span>, <span class="st">"feces"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>In this table, the Binomial Test p-value corresponds to the null hypothesis</p>
<p>H0: the proportion of signatures in which the taxon is reported increased or decreased, relative to the total number of signatures in which it is reported, is equal to 0.5</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span><span class="fu">bugSigSimple</span><span class="fu">::</span><span class="fu"><a href="../reference/createTaxonTable.html">createTaxonTable</a></span><span class="op">(</span><span class="va">gut_sigs</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
Taxon Name
</th>
<th style="text-align:left;">
Taxonomic Level
</th>
<th style="text-align:right;">
total_signatures
</th>
<th style="text-align:right;">
increased_signatures
</th>
<th style="text-align:right;">
decreased_signatures
</th>
<th style="text-align:right;">
Binomial Test pval
</th>
<th style="text-align:left;">
kingdom
</th>
<th style="text-align:left;">
phylum
</th>
<th style="text-align:left;">
class
</th>
<th style="text-align:left;">
order
</th>
<th style="text-align:left;">
family
</th>
<th style="text-align:left;">
genus
</th>
<th style="text-align:left;">
species
</th>
<th style="text-align:right;">
n_signatures
</th>
<th style="text-align:left;">
metaphlan_name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Bifidobacterium catenulatum
</td>
<td style="text-align:left;">
species
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.130
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Actinobacteria
</td>
<td style="text-align:left;">
Actinomycetia
</td>
<td style="text-align:left;">
Bifidobacteriales
</td>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
Bifidobacterium
</td>
<td style="text-align:left;">
Bifidobacterium catenulatum
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
k__Bacteria|p__Actinobacteria|c__Actinomycetia|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium catenulatum
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides
</td>
<td style="text-align:left;">
genus
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Bacteroidetes
</td>
<td style="text-align:left;">
Bacteroidia
</td>
<td style="text-align:left;">
Bacteroidales
</td>
<td style="text-align:left;">
Bacteroidaceae
</td>
<td style="text-align:left;">
Bacteroides
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Bacteroidaceae|g__Bacteroides
</td>
</tr>
<tr>
<td style="text-align:left;">
Enterobacteriaceae
</td>
<td style="text-align:left;">
family
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.130
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Proteobacteria
</td>
<td style="text-align:left;">
Gammaproteobacteria
</td>
<td style="text-align:left;">
Enterobacterales
</td>
<td style="text-align:left;">
Enterobacteriaceae
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Enterobacterales|f__Enterobacteriaceae
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacterium adolescentis
</td>
<td style="text-align:left;">
species
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.690
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Actinobacteria
</td>
<td style="text-align:left;">
Actinomycetia
</td>
<td style="text-align:left;">
Bifidobacteriales
</td>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
Bifidobacterium
</td>
<td style="text-align:left;">
Bifidobacterium adolescentis
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
k__Bacteria|p__Actinobacteria|c__Actinomycetia|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium adolescentis
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacterium longum
</td>
<td style="text-align:left;">
species
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.690
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Actinobacteria
</td>
<td style="text-align:left;">
Actinomycetia
</td>
<td style="text-align:left;">
Bifidobacteriales
</td>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
Bifidobacterium
</td>
<td style="text-align:left;">
Bifidobacterium longum
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
k__Bacteria|p__Actinobacteria|c__Actinomycetia|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium longum
</td>
</tr>
<tr>
<td style="text-align:left;">
Blautia
</td>
<td style="text-align:left;">
genus
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Firmicutes
</td>
<td style="text-align:left;">
Clostridia
</td>
<td style="text-align:left;">
Eubacteriales
</td>
<td style="text-align:left;">
Lachnospiraceae
</td>
<td style="text-align:left;">
Blautia
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
k__Bacteria|p__Firmicutes|c__Clostridia|o__Eubacteriales|f__Lachnospiraceae|g__Blautia
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridioides difficile
</td>
<td style="text-align:left;">
species
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.031
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Firmicutes
</td>
<td style="text-align:left;">
Clostridia
</td>
<td style="text-align:left;">
Eubacteriales
</td>
<td style="text-align:left;">
Peptostreptococcaceae
</td>
<td style="text-align:left;">
Clostridioides
</td>
<td style="text-align:left;">
Clostridioides difficile
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
k__Bacteria|p__Firmicutes|c__Clostridia|o__Eubacteriales|f__Peptostreptococcaceae|g__Clostridioides|s__Clostridioides difficile
</td>
</tr>
<tr>
<td style="text-align:left;">
Veillonella
</td>
<td style="text-align:left;">
genus
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.690
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Firmicutes
</td>
<td style="text-align:left;">
Negativicutes
</td>
<td style="text-align:left;">
Veillonellales
</td>
<td style="text-align:left;">
Veillonellaceae
</td>
<td style="text-align:left;">
Veillonella
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
k__Bacteria|p__Firmicutes|c__Negativicutes|o__Veillonellales|f__Veillonellaceae|g__Veillonella
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
family
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Actinobacteria
</td>
<td style="text-align:left;">
Actinomycetia
</td>
<td style="text-align:left;">
Bifidobacteriales
</td>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:left;">
k__Bacteria|p__Actinobacteria|c__Actinomycetia|o__Bifidobacteriales|f__Bifidobacteriaceae
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacterium bifidum
</td>
<td style="text-align:left;">
species
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.630
</td>
<td style="text-align:left;">
Bacteria
</td>
<td style="text-align:left;">
Actinobacteria
</td>
<td style="text-align:left;">
Actinomycetia
</td>
<td style="text-align:left;">
Bifidobacteriales
</td>
<td style="text-align:left;">
Bifidobacteriaceae
</td>
<td style="text-align:left;">
Bifidobacterium
</td>
<td style="text-align:left;">
Bifidobacterium bifidum
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
k__Bacteria|p__Actinobacteria|c__Actinomycetia|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium bifidum
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="cluster-analysis">Cluster analysis<a class="anchor" aria-label="anchor" href="#cluster-analysis"></a>
</h2>
<p>Note, this EDA should really be done <em>before</em> hypothesis testing.</p>
<p>First calculate pairwise overlaps for all signatures of length &gt; 1:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">allsigs</span> <span class="op">&lt;-</span> <span class="fu">bugsigdbr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bugsigdbr/man/getSignatures.html" class="external-link">getSignatures</a></span><span class="op">(</span><span class="va">dat_condition</span>, tax.id.type <span class="op">=</span> <span class="st">"taxname"</span><span class="op">)</span>
<span class="va">allsigs</span> <span class="op">&lt;-</span> <span class="va">allsigs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">allsigs</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span> <span class="co">#require length &gt; 1</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">allsigs</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 32</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mydists</span> <span class="op">&lt;-</span> <span class="fu">BugSigDBStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BugSigDBStats/man/calcPairwiseOverlaps.html" class="external-link">calcPairwiseOverlaps</a></span><span class="op">(</span><span class="va">allsigs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">mydists</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 126   8</span></code></pre>
<p>What is the distribution of signature lengths?</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">siglengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">allsigs</span>, <span class="va">length</span><span class="op">)</span>
<span class="va">siglengths.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>siglengths <span class="op">=</span> <span class="va">siglengths</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">siglengths.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">siglengths</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="capstoneanalysis_kweku_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">siglengths</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## siglengths</span>
<span class="co">##  2  3  4  5  6  8  9 11 13 14 15 16 17 18 19 </span>
<span class="co">##  7  5  3  2  2  1  2  2  1  1  1  2  1  1  1</span></code></pre>
<p>Create a matrix of Jaccard similarities (0 for no overlap, 1 for 100% overlap)</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">jmat</span> <span class="op">&lt;-</span> <span class="fu">BugSigDBStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BugSigDBStats/man/calcJaccardSimilarity.html" class="external-link">calcJaccardSimilarity</a></span><span class="op">(</span><span class="va">allsigs</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: grid</span></code></pre>
<pre><code><span class="co">## ========================================</span>
<span class="co">## ComplexHeatmap version 2.10.0</span>
<span class="co">## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span>
<span class="co">## Github page: https://github.com/jokergoo/ComplexHeatmap</span>
<span class="co">## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span>
<span class="co">## </span>
<span class="co">## If you use it in published research, please cite:</span>
<span class="co">## Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span>
<span class="co">##   genomic data. Bioinformatics 2016.</span>
<span class="co">## </span>
<span class="co">## The new InteractiveComplexHeatmap package can directly export static </span>
<span class="co">## complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span>
<span class="co">## </span>
<span class="co">## This message can be suppressed by:</span>
<span class="co">##   suppressPackageStartupMessages(library(ComplexHeatmap))</span>
<span class="co">## ========================================</span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ha</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html" class="external-link">HeatmapAnnotation</a></span><span class="op">(</span>`Signature Length` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_barplot.html" class="external-link">anno_barplot</a></span><span class="op">(</span><span class="va">siglengths</span><span class="op">)</span><span class="op">)</span>
<span class="va">hr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>`Signature Length` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_barplot.html" class="external-link">anno_barplot</a></span><span class="op">(</span><span class="va">siglengths</span><span class="op">)</span><span class="op">)</span>
<span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span>
  <span class="va">jmat</span>,
  top_annotation <span class="op">=</span> <span class="va">ha</span>, left_annotation <span class="op">=</span> <span class="va">hr</span>,
  row_names_max_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"cm"</span><span class="op">)</span>,
  column_names_max_height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"cm"</span><span class="op">)</span>,
<span class="co">#  row_labels = sub(".+:", "", rownames(jmat)),  #get rid of study labels</span>
  column_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".+:"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">jmat</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">hm</span></code></pre></div>
<p><img src="capstoneanalysis_kweku_files/figure-html/unnamed-chunk-11-1.png" width="1920"></p>
<p>Use this interactively to make an interactive heatmap. Some expanding of the default size is required to see anything. Creating a sub-heatmap, then exporting it as a table, allows in-depth identification of the subgroups.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/InteractiveComplexHeatmap" class="external-link">InteractiveComplexHeatmap</a></span><span class="op">)</span>
<span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">hm</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/InteractiveComplexHeatmap/man/htShiny.html" class="external-link">htShiny</a></span><span class="op">(</span><span class="va">hm</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="va">jmat</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span></code></pre></div>
<p><img src="capstoneanalysis_kweku_files/figure-html/unnamed-chunk-13-1.png" width="1920"></p>
<p>This tree can be cut to show the clusters, for example. The clusters of more than 1 signature but less than ~10 signatures are most likely to be something interesting.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clusts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cutree.html" class="external-link">cutree</a></span><span class="op">(</span><span class="va">hc</span>, k <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span>  <span class="co">#set the number of clusters here with k</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">clusts</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusts</span><span class="op">)</span><span class="op">[</span><span class="va">clusts</span> <span class="op">==</span> <span class="va">i</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [[1]]</span>
<span class="co">##  [1] "bsdb:192/1/1_irritable-bowel-syndrome:diarrhea-predominant-irritable-bowel-syndrome_vs_healthy-controls_UP"</span>
<span class="co">##  [2] "bsdb:192/2/1_irritable-bowel-syndrome:diarrhea-predominant-irritable-bowel-syndrome_vs_healthy-controls_UP"</span>
<span class="co">##  [3] "bsdb:443/3/2_irritable-bowel-syndrome:IBS-D_vs_Health-Control_DOWN"                                        </span>
<span class="co">##  [4] "bsdb:488/1/1_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_UP"                                         </span>
<span class="co">##  [5] "bsdb:488/1/2_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_DOWN"                                       </span>
<span class="co">##  [6] "bsdb:491/1/1_irritable-bowel-syndrome:IBS_vs_Healthy-Control_UP"                                           </span>
<span class="co">##  [7] "bsdb:491/1/2_irritable-bowel-syndrome:IBS_vs_Healthy-Control_DOWN"                                         </span>
<span class="co">##  [8] "bsdb:499/1/2_irritable-bowel-syndrome:IBS-A_vs_Healthy-Control_UP"                                         </span>
<span class="co">##  [9] "bsdb:499/2/2_irritable-bowel-syndrome:IBS-C_vs_Healthy-Control_UP"                                         </span>
<span class="co">## [10] "bsdb:499/3/2_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_UP"                                         </span>
<span class="co">## [11] "bsdb:499/4/2_irritable-bowel-syndrome:IBS-(All-patients)_vs_Healthy-Control_UP"                            </span>
<span class="co">## [12] "bsdb:502/1/1_irritable-bowel-syndrome:IBS_vs_Healthy-Control_UP"                                           </span>
<span class="co">## [13] "bsdb:502/2/1_irritable-bowel-syndrome:IBS-C_vs_Healthy-Control_UP"                                         </span>
<span class="co">## [14] "bsdb:502/3/1_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_UP"                                         </span>
<span class="co">## [15] "bsdb:502/4/1_irritable-bowel-syndrome:IBS-M_vs_Healthy-Control_UP"                                         </span>
<span class="co">## [16] "bsdb:505/1/1_irritable-bowel-syndrome:IBS_vs_Healthy-Control_UP"                                           </span>
<span class="co">## </span>
<span class="co">## [[2]]</span>
<span class="co">## [1] "bsdb:443/1/1_irritable-bowel-syndrome:IBS_vs_Health-Control_UP"    </span>
<span class="co">## [2] "bsdb:443/2/1_irritable-bowel-syndrome:IBS-C_vs_Health-Control_UP"  </span>
<span class="co">## [3] "bsdb:443/4/1_irritable-bowel-syndrome:IBS-U_vs_Health-Control_UP"  </span>
<span class="co">## [4] "bsdb:443/4/2_irritable-bowel-syndrome:IBS-U_vs_Health-Control_DOWN"</span>
<span class="co">## </span>
<span class="co">## [[3]]</span>
<span class="co">## [1] "bsdb:443/3/1_irritable-bowel-syndrome:IBS-D_vs_Health-Control_UP"</span>
<span class="co">## </span>
<span class="co">## [[4]]</span>
<span class="co">## [1] "bsdb:499/1/1_irritable-bowel-syndrome:IBS-A_vs_Healthy-Control_DOWN"             </span>
<span class="co">## [2] "bsdb:499/2/1_irritable-bowel-syndrome:IBS-C_vs_Healthy-Control_DOWN"             </span>
<span class="co">## [3] "bsdb:499/3/1_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_DOWN"             </span>
<span class="co">## [4] "bsdb:499/4/1_irritable-bowel-syndrome:IBS-(All-patients)_vs_Healthy-Control_DOWN"</span>
<span class="co">## </span>
<span class="co">## [[5]]</span>
<span class="co">## [1] "bsdb:502/1/2_irritable-bowel-syndrome:IBS_vs_Healthy-Control_DOWN"  </span>
<span class="co">## [2] "bsdb:502/2/2_irritable-bowel-syndrome:IBS-C_vs_Healthy-Control_DOWN"</span>
<span class="co">## </span>
<span class="co">## [[6]]</span>
<span class="co">## [1] "bsdb:502/3/2_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_DOWN"</span>
<span class="co">## </span>
<span class="co">## [[7]]</span>
<span class="co">## [1] "bsdb:505/1/2_irritable-bowel-syndrome:IBS_vs_Healthy-Control_DOWN"  </span>
<span class="co">## [2] "bsdb:505/2/2_irritable-bowel-syndrome:IBS-D_vs_Healthy-Control_DOWN"</span>
<span class="co">## [3] "bsdb:505/3/2_irritable-bowel-syndrome:IBS-A_vs_Healthy-Control_DOWN"</span>
<span class="co">## </span>
<span class="co">## [[8]]</span>
<span class="co">## [1] "bsdb:505/4/2_irritable-bowel-syndrome:IBS-C_vs_Healthy-Control_DOWN"</span></code></pre>
</div>
<div class="section level2">
<h2 id="create-a-wide-format-dataframe">Create a wide-format dataframe<a class="anchor" aria-label="anchor" href="#create-a-wide-format-dataframe"></a>
</h2>
<p>This would be suitable for regression analysis.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_withsigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dat_condition</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">dat_condition</span><span class="op">$</span><span class="va">`NCBI Taxonomy IDs`</span><span class="op">)</span><span class="op">)</span>
<span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu">bugsigdbr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bugsigdbr/man/getSignatures.html" class="external-link">getSignatures</a></span><span class="op">(</span><span class="va">dat_withsigs</span>, tax.id.type <span class="op">=</span> <span class="st">"taxname"</span><span class="op">)</span>
<span class="va">cmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu">safe</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/safe/man/getCmatrix.html" class="external-link">getCmatrix</a></span><span class="op">(</span><span class="va">sigs</span>, as.matrix <span class="op">=</span> <span class="cn">TRUE</span>, min.size <span class="op">=</span> <span class="fl">0</span>, prune <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## WARNING: rows are sorted elements of keyword.list</span>
<span class="co">## 40 categories formed</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">cmat</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">cdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">dat_withsigs</span>, <span class="va">cdf</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cdf</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">54</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">##  [1] "Study"                      "Study design"              </span>
<span class="co">##  [3] "PMID"                       "DOI"                       </span>
<span class="co">##  [5] "URL"                        "Authors"                   </span>
<span class="co">##  [7] "Title"                      "Journal"                   </span>
<span class="co">##  [9] "Year"                       "Experiment"                </span>
<span class="co">## [11] "Location of subjects"       "Host species"              </span>
<span class="co">## [13] "Body site"                  "UBERON ID"                 </span>
<span class="co">## [15] "Condition"                  "EFO ID"                    </span>
<span class="co">## [17] "Group 0 name"               "Group 1 name"              </span>
<span class="co">## [19] "Group 1 definition"         "Group 0 sample size"       </span>
<span class="co">## [21] "Group 1 sample size"        "Antibiotics exclusion"     </span>
<span class="co">## [23] "Sequencing type"            "16S variable region"       </span>
<span class="co">## [25] "Sequencing platform"        "Statistical test"          </span>
<span class="co">## [27] "Significance threshold"     "MHT correction"            </span>
<span class="co">## [29] "LDA Score above"            "Matched on"                </span>
<span class="co">## [31] "Confounders controlled for" "Pielou"                    </span>
<span class="co">## [33] "Shannon"                    "Chao1"                     </span>
<span class="co">## [35] "Simpson"                    "Inverse Simpson"           </span>
<span class="co">## [37] "Richness"                   "Signature page name"       </span>
<span class="co">## [39] "Source"                     "Curated date"              </span>
<span class="co">## [41] "Curator"                    "Revision editor"           </span>
<span class="co">## [43] "Description"                "Abundance in Group 1"      </span>
<span class="co">## [45] "MetaPhlAn taxon names"      "NCBI Taxonomy IDs"         </span>
<span class="co">## [47] "State"                      "Reviewer"                  </span>
<span class="co">## [49] "comparison1"                "[Clostridium] cellulosi"   </span>
<span class="co">## [51] "[Clostridium] leptum"       "[Clostridium] symbiosum"   </span>
<span class="co">## [53] "[Eubacterium] rectale"      "[Ruminococcus] gnavus"</span></code></pre>
<p>Note this has a number of columns that are mostly zeros, it could be filtered significantly for any regression or machine learning analysis:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cdf</span><span class="op">[[</span><span class="st">"Bifidobacterium catenulatum"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  0  1 </span>
<span class="co">## 33  7</span></code></pre>
<p>Create another heatmap on correlations of presence/absence of taxa. This is not necessary because the previous Jaccard Index heatmap is probably better, it is just a demonstration of doing something with the taxa presence/absence directly.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sigcors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">cmat</span><span class="op">)</span><span class="op">)</span>
<span class="va">siglengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">sigs</span>, <span class="va">length</span><span class="op">)</span>
<span class="va">ha</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html" class="external-link">HeatmapAnnotation</a></span><span class="op">(</span>`Signature Length` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_barplot.html" class="external-link">anno_barplot</a></span><span class="op">(</span><span class="va">siglengths</span><span class="op">)</span><span class="op">)</span>
<span class="va">hr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>`Signature Length` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_barplot.html" class="external-link">anno_barplot</a></span><span class="op">(</span><span class="va">siglengths</span><span class="op">)</span><span class="op">)</span>
<span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span>
  <span class="va">sigcors</span>,
  top_annotation <span class="op">=</span> <span class="va">ha</span>, left_annotation <span class="op">=</span> <span class="va">hr</span>,
  row_names_max_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"cm"</span><span class="op">)</span>,
  column_names_max_height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">20</span>, <span class="st">"cm"</span><span class="op">)</span>,
 <span class="co"># row_labels = sub(".+:", "", rownames(sigcors)), ##removing study just to make signature names legible</span>
  column_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".+:"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sigcors</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">hm</span></code></pre></div>
<p><img src="capstoneanalysis_kweku_files/figure-html/unnamed-chunk-17-1.png" width="1920"></p>
<p>Use this interactively to make an interactive heatmap:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/InteractiveComplexHeatmap" class="external-link">InteractiveComplexHeatmap</a></span><span class="op">)</span>
<span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">hm</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/InteractiveComplexHeatmap/man/htShiny.html" class="external-link">htShiny</a></span><span class="op">(</span><span class="va">hm</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
